### PR TITLE
Update build

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

**Comment by cchantep** [comment]
Akka-Stream is published for Scala 3, but not Akka-HTTP, and due to inter-dependency, one cannot be `_2.13` and the other `_3`, otherwise raise class loading issue.

```
[error]  org.specs2.specification.core.FatalExecution: akka.actor.ExtensionId.$init$(Lakka/actor/ExtensionId;)V (Http.scala:845)
[error] akka.http.scaladsl.Http$.<clinit>(Http.scala:845)
``

**Comment by xuwei-k** [comment]
disable integration-tests if Scala 3 🤔 

```diff
diff --git a/build.sbt b/build.sbt
index ea2426a..03a2033 100644
--- a/build.sbt
+++ b/build.sbt
@@ -393,6 +393,15 @@ lazy val `integration-tests` = project
   .settings(publish / skip := true)
   .settings(
     Test / fork := true,
+    Test / sources := {
+      if (scalaBinaryVersion.value == "3") {
+        // TODO remove if akka-http for Scala 3 released
+        // https://github.com/akka/akka-http/issues/3891
+        Nil
+      } else {
+        (Test / sources).value
+      }
+    },
     concurrentRestrictions += Tags.limitAll(1), // only one integration test at a time
```

**Comment by cchantep** [comment]
> disable integration-tests if Scala 3 🤔
> 
> ```diff
> diff --git a/build.sbt b/build.sbt
> index ea2426a..03a2033 100644
> --- a/build.sbt
> +++ b/build.sbt
> @@ -393,6 +393,15 @@ lazy val `integration-tests` = project
>    .settings(publish / skip := true)
>    .settings(
>      Test / fork := true,
> +    Test / sources := {
> +      if (scalaBinaryVersion.value == "3") {
> +        // TODO akka-http for Scala 3
> +        // https://github.com/akka/akka-http/issues/3891
> +        Nil
> +      } else {
> +        (Test / sources).value
> +      }
> +    },
>      concurrentRestrictions += Tags.limitAll(1), // only one integration test at a time
> ```

Rather wait `akka-http`

**Comment by jtjeferreira** [comment]
since `akka-http` with scala3 will only be released with BSL License, how can we move this forward? using the BSL version just for the tests is okay?

File: project/Dependencies.scala [hunk 1]
```diff
@@ -52,8 +52,8 @@ object Dependencies {
 
   val asyncHttpClient = Seq("org.asynchttpclient" % "async-http-client" % "2.12.3")
 
-  val akkaStreams = Seq(("com.typesafe.akka" %% "akka-stream" % "2.6.18").cross(CrossVersion.for3Use2_13))
-  val akkaHttp    = Seq(("com.typesafe.akka" %% "akka-http"   % "10.2.7").cross(CrossVersion.for3Use2_13))
+  val akkaStreams = Seq("com.typesafe.akka" %% "akka-stream" % "2.6.18")
+  val akkaHttp    = Seq(("com.typesafe.akka" %% "akka-http" % "10.2.7").cross(CrossVersion.for3Use2_13))
```
Comment (jtjeferreira) [inline project/Dependencies.scala:56]:
```suggestion
  val akkaHttp    = Seq("com.typesafe.akka" %% "akka-http" % "10.5.0-M1")
```

**Pull request merged**